tests/test_event_detection.py

```diff
@@ -1,4 +1,120 @@
+'''Code Tests'''
+
 # Authors: Ryan Thorpe <ryvthorpe@gmail.com>
+# Authors: Darcy Diesburg <darcy.diesburg@gmail.com>
+
+import os.path as op
+
+import numpy as np
+from scipy.io import loadmat
+
+import SpectralEvents.spectralevents as se
+
 
 def test_event_comparison():
+    '''Test if the output from MATLAB and Python event detection scripts align
+
+    Parameters
+    ----------
+    method : float
+    The event detection method (1, 2, or 3) used to detect spectral events.
+
+    Notes
+    -----
+    Currently only supports testing method 1 and 3, with method 2 to be added
+    to the Python event detection functions in the future. Python output is
+    tested against the output of MATLAB scripts from Shin et al., 2017.
+    Detection of the same number of beta events within each trial is required,
+    and tolerable margins of deviation for the other parameters.
+    '''
+
+    # Define tolerable deviance for spectral event characteristics
+    dev_max_time = 5  # +-5ms allowed for timing latency diff
+    dev_count_avg = 0.001  # 1/1000 trials, on average
+
+    # Load MATLAB data
+    root_dir = op.dirname(se.__file__)
+    matlab_ev_struct = loadmat(op.join(root_dir, 'tests',
+                                       'beta_events_shin_2017.mat'))
+
+    # Get python events by running find_events on demo data
+    subj_ids = [str(id) for id in range(1, 10 + 1)]  # subject IDs 1-10
+
+    # set parameters
+    samp_freq = 600
+    n_times = 600
+    # fequency values (Hz) over which to calculate TFR
+    freqs = list(range(1, 60 + 1))
+    times = np.arange(n_times) / samp_freq  # seconds
+    event_band = [15, 29]  # beta band (Hz)
+    thresh_fom = 6.0  # factor-of-the-median threshold
+
+    # get python evs by running find_events()
+    py_ev_dict = list()
+    for _, id_val in enumerate(subj_ids):
+        fname = op.join(
+            root_dir, 'data', 'prestim_humandetection_600hzMEG_subject' +
+            id_val + '.mat')
+        raw_data = loadmat(fname)
+        data = raw_data['prestim_raw_yes_no']
+        # calculate TFR
+        tfrs = se.tfr(data, freqs, samp_freq)
+
+        # find events
+        subj_evs = se.find_events(tfr=tfrs, times=times, freqs=freqs,
+                                  event_band=event_band, thresholds=None,
+                                  threshold_FOM=thresh_fom)
+        py_ev_dict.append(subj_evs)
+
+    # take ev counts and timing latencies from respective dictionaries
+    # and structures to put into a subj x trial matrix for comparison
+    py_ev_count_mat = np.zeros((10, 200))
+    for subj_idx, id_val in enumerate(subj_ids):
+        py_ev_count_mat[subj_idx, :] = np.array(
+            [len(trial) for trial in py_ev_dict[subj_idx]])
```

Comment:
This doesn't actually need to be converted into a `ndarray` object. [Zen of Python](https://peps.python.org/pep-0020/): simple is better than complex